src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -434,6 +481,8 @@ private void InitializeCore()
 
             DefineFullCategoryNames(LibraryRootCategories, "");
             InsertClassesIntoTree(LibraryRootCategories);
+
+            useDebouncer = DynamoModel.FeatureFlags?.CheckFeatureFlag("searchbar_debounce", false) ?? false;
```

Comment:
I am not sure how often this class is instantiated, or if it's only once during dynamo view model creation.
If it's frequent, then I think this use of `CheckFeatureFlag` is safe - if it's only once at startup then I think it will be more predictable to either:
1. check the feature flag onTextChanged, it's probably safe to cache at that point, or don't if it does not affect performance.
2. use the static `DynamoFeatureFlagsManager.FlagsRetrieved` event to make the check only after we have received flags.... unfortunately, If you try to use that from this class's constructor - I think you could get into a situation where you then subscribe this event too late, ideally it would be done before the feature flags retrieval is made. If you want to only make this check once you could lump it into here: https://github.com/DynamoDS/Dynamo/blob/23bf46389b41d9873a33bb31eafbd4310ad8ee86/src/DynamoCore/Models/DynamoModel.cs#L1015


It's not obvious, but `CheckFeatureFlag` will only check the local cache of flags which may not have been retrieved yet from the flags service.


Personally, I would go for 1 because it's simpler to reason IMO about, touches less files etc unless performance is an issue but under the hood it's just a dictionary lookup.